### PR TITLE
ISSUE-37 improve precision of ShannonEntropyFactory

### PR DESCRIPTION
**Pull request by ememisya**
I've added a couple minor changes:

- Dictionary should not be instantiated with a null.
- Dictionary interface should return its size.

Reason for those changes was to ensure that we don't flag a Shannon entropy estimate with a supplied empty dictionary.

PS: I love a resolution where there are more deletions than additions :+1:

### CODE REVIEW

File: src/main/java/org/passay/entropy/ShannonEntropyFactory.java [hunk 1]
```diff
@@ -38,57 +56,26 @@ public static ShannonEntropy createEntropy(final List<Rule> passwordRules, final
...
-   * @param  rule  to validate for the matching composition rule
-   * @param  characters  to check for composition
-   * @param  passwordData  with the contents of the password.
+   * @param  passwordData  to check for composition
    *
-   * @return  whether the supplied rule has the correct composition
+   * @return  true if valid, false otherwise
    */
-  private static boolean hasComposition(final Rule rule, final String characters, final PasswordData passwordData)
+  protected static boolean hasComposition(final PasswordData passwordData)
```
Comment (serac) [inline src/main/java/org/passay/entropy/ShannonEntropyFactory.java]: Changes to accessibility typically sound a warning bell. Why this change?

**Comment by serac** [comment]
Looks good generally. One minor question.

File: src/main/java/org/passay/entropy/ShannonEntropyFactory.java (hunk 1 above)
Comment (ememisya) [inline src/main/java/org/passay/entropy/ShannonEntropyFactory.java]: Good catch, I actually had a class I did not end up including the commit, we can revert that to private.
